src/classes/CallableApiParameters.cls

```diff
@@ -326,6 +327,36 @@ public class CallableApiParameters {
         return null;
     }
 
+    public Map<Id, RD2_ApiService.PauseObject> getPauseObjectsById(String paramName) {
+        String paramType = 'Map<Id, RD2_ApiService.PauseObject>';
+        validateRequiredParameter(paramType, paramName);
+        try {
+            return convertObjectIntoPauseObject((Map<Id, Object>) params.get(paramName));
+        } catch (Exception ex) {
+            throwParameterException('Map<Id, RD2_ApiService.PauseObject>', paramName);
+        }
+
+        return null;
+    }
+
+    private Map<Id, RD2_ApiService.PauseObject> convertObjectIntoPauseObject(Map<Id, Object> pauseParams) {
+        Map<Id, RD2_ApiService.PauseObject> pauseData = new Map<Id, RD2_ApiService.PauseObject>();
+
+        for (Id key : pauseParams.keySet()) {
+            Object pauseObject = pauseParams.get(key);
+
+            if (!(pauseObject instanceof String)) {
+                pauseObject = JSON.serialize(pauseObject);
+            }
+            RD2_ApiService.PauseObject convertedPauseStructure = 
+                (RD2_ApiService.PauseObject) JSON.deserialize((String) pauseObject, RD2_ApiService.PauseObject.class);
+
+            pauseData.put(key, convertedPauseStructure);
+        }
```

Comment:
Salesforce cannot convert user created object into a specific class without serializing the object. Line 348 does a quick check to see if the object is already serialize.